src/app/components/Home/Office/index.tsx

```diff
@@ -1,47 +1,85 @@
-import React, { useEffect } from "react";
+import React, { useEffect, useState } from "react";
 import { useDispatch, useSelector } from "react-redux";
-import { useParams } from "react-router-dom";
+import { Redirect, useParams } from "react-router-dom";
 
 // components
-import Floors from "./Floors";
+import Loader from "../../Loader";
+import NavOnFloors from "./NavOnFloors";
 import Canvas from "./Canvas";
 
 // styles
 import { HeaderOffice, TitleOffice } from "./components";
 
 // scelectors
-import { getFloorsData } from "../../../redux/selectors";
+import {
+  getAuthUser,
+  getLoadingState,
+  getSelectedOffice,
+} from "../../../redux/selectors";
 
 // actions
 import {
-  fetchFloorsData,
+  fetchSelectedOffice,
   setSelectedFloor,
 } from "../../../redux/actions/selectOffice";
 
+// types
+import { Addres, Floors, SelectedOffice } from "../../../services/BD/type";
+
 const Office: React.FC = () => {
-  const { officeId } = useParams<{ officeId: string }>();
+  const user = useSelector(getAuthUser);
+  const loading = useSelector(getLoadingState);
   const dispatch = useDispatch();
-  const floors = useSelector(getFloorsData);
-  const numberOfFloor = Object.values(floors).map((i: any) => i.number);
-  if (numberOfFloor.length > 0) {
-    dispatch(setSelectedFloor(1));
-  }
+  const { officeId } = useParams<{ officeId: string }>();
+  const selectedOffice: SelectedOffice = useSelector(getSelectedOffice);
+  const [selectedOfficeAddres, setSelectedOfficeAddres] = useState<Addres>({
+    country: "",
+    city: "",
+    location: "",
+  });
+  const [dataFloors, setDataFloors] = useState<Floors>([]);
 
   useEffect(() => {
-    dispatch(fetchFloorsData(officeId));
+    if (user) {
+      dispatch(fetchSelectedOffice(officeId));
+    }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [officeId]);
+  }, [user]);
+
+  useEffect(() => {
+    if (selectedOffice) {
+      setSelectedOfficeAddres({
+        country: selectedOffice.addres.country,
+        city: selectedOffice.addres.city,
+        location: selectedOffice.addres.location,
+      });
+      setDataFloors(selectedOffice.floors);
+      if (selectedOffice.floors.length > 0) {
+        dispatch(setSelectedFloor(1));
+      }
+    }
+  }, [dispatch, selectedOffice]);
```

Comment:
You can read office address and floors data directly from `selectedOffice`, no need to set state. Also initial selected floor  number can be set in saga or reducer once data is available.